src/app/modules/core/components/sidebar/sidebar.component.ts

```diff
@@ -75,12 +84,31 @@ export class SidebarComponent {
       disabled: true,
       mode: 'indeterminate'
     }
+
+    this.MapService.LayersControl.subscribe(data => {
+      if (this.overlays.length > 0 || this.baselayers.length > 0) {
+        this.overlays = []
+        this.baselayers = []
+      }
+      this.overlays = data.overlays;
+      this.baselayers = data.baseLayers;
+    })
+
+    this.model = {
+      baselayers: {},
+      overlays: {}
+    };
+
    }
 
+  private SetAll(LayerName: string, kind:string) {
```

Comment:
A few things here, 'Private' is a key word that indicates it is only accessible internal to the class/component, since it is being called by html it should be public.

As mentioned in the html we should probably break this apart to 2 methods. one for baselayers, and another for overlays... In the future we may want to handle them differently.